objects/src/accounts/mod.rs

```diff
@@ -94,6 +98,20 @@ impl Account {
         Self { id, vault, storage, code, nonce }
     }
 
+    // TODO: Document.
+    pub fn from_components(
+        seed: Word,
+        account_type: AccountType,
+        components: &[AccountComponent],
+    ) -> Result<Self, AccountError> {
```

Comment:
This will replace `Account::new()` right?

Also, a couple of thoughts:
- If we have a way to infer account type from the underlying components (i.e., if it is a faucet or not), we can avoid passing it as a parameter.
- If we do pass account type as a parameter, we need to add a check to make sure the generated ID matches this type.

Reply:
It was still necessary to pass `account_type`, unfortunately. Because using the `BTreeSet<AccountType>` on account components to specify support, we cannot infer the type from that afaict. I.e. if we instantiate an account from `RpoFalcon512` which supports all types, which type would we infer? I guess the default type "Regular Account Updatable Code" would be reasonable, but it seems better to pass it explicitly.

Maybe there is some better way to do it, but I think for now passing the AccountType is not too bad.

Because of your second point I think there is some more thought needed to figure out how to prevent someone from creating an `Account` with `AccountCode` and `AccountStorage` which was created for a different account type than the passed `seed` turns into. Ideally this can be addressed in a separate PR.